moped-editor/src/queries/tags.js

```diff
@@ -0,0 +1,40 @@
+import { gql } from "@apollo/client";
+
+export const TAGS_QUERY = gql`
+  query TagsQuery($projectId: Int!) {
+    moped_proj_tags(
+      where: { project_id: { _eq: $projectId }, is_deleted: { _eq: false } }
+    ) {
+      id
+      moped_tag {
+        name
+        id
+      }
+    }
+    moped_tags(order_by: { name: asc }) {
+      id
+      name
+    }
+  }
+`;
+
+export const DELETE_PROJECT_TAG = gql`
+  mutation DeleteProjectTag($id: Int!) {
+    update_moped_proj_tags_by_pk(
+      pk_columns: { id: $id }
+      _set: { is_deleted: true }
+    ) {
+      id
+    }
+  }
+`;
+
+export const ADD_PROJECT_TAGS = gql`
+  mutation AddProjectTags($objects: [moped_proj_tags_insert_input!]!) {
+    insert_moped_proj_tags(objects: $objects) {
+      returning {
+        id
+      }
+    }
+  }
+`;
```

Comment:
Just thinking through how the add/delete works here. 1. User adds a tag (`is_deleted` = false) 2. User deletes that tag (`is_deleted` = true) 3. Another user adds the same tag and a new `moped_proj_tags` record is created with `is_deleted` = false

Over time, `moped_proj_tags` records stack up in the table. Not saying this is how it should or shouldn't work - just making sure this is how it is intended to work. My brain is still waking up 🧠 😅 
<img width="793" alt="Screen Shot 2022-09-13 at 1 20 38 PM" src="https://user-images.githubusercontent.com/37249039/189980274-956911e0-7af8-4f90-bb22-13090c50dc62.png">

Reply:
thanks — i think that's fine. since our feature is querying this table by `project_id`, and the table is going get rather larger, maybe we should index that `project_id` column?